### ISSUE TITLE
Efficient conversion functions between Set and Map

### ISSUE THREAD

**Issue by np** [purescript/purescript-ordered-collections#20]
Since the implementation of `Set` is based on `Map` it would be nice to expose conversions functions. In this case these are only the constructor and projection of the newtype.

**Comment by kritzcreek** [comment]
The conversion function from `Map` to `Set` is just the `keys` function. Going from `Set k` to `Map k Unit` doesn't seem terribly useful?

**Comment by np** [comment]
I agree that `keys` does the first half. However, currently it uses `unsafeCoerce` and this could be avoided.

Now `toMap` can be useful for performance reasons since the functor instances can keep the tree structure unchanged.

Consider for instance that you want to create a map from a set and a function:

```
annotateSet :: forall a b. (a -> b) -> Set a -> Map a b
annotateSet f = mapWithIndex (\k _ -> f k) <<< toMap
```

This version requires a single traversal and there is no comparisons done on the elements (no `Ord` required).

**Comment by kritzcreek** [comment]
Good example, I wouldn't mind adding `toMap` to `Data.Set` in that case.

 I don't really care about the use of `unsafeCoerce` for `keys`, but if you can implement the function just as quick without it please go ahead.

**Comment by JordanMartinez** [comment]
Closed by #31
